cpp/src/round/round.cu

```diff
@@ -209,6 +227,49 @@ std::unique_ptr<column> round_with(column_view const& input,
   return result;
 }
 
+template <typename T,
+          template <typename>
+          typename RoundFunctor,
+          typename std::enable_if_t<cudf::is_fixed_point<T>()>* = nullptr>
+std::unique_ptr<column> round_with(column_view const& input,
+                                   int32_t decimal_places,
+                                   cudaStream_t stream,
+                                   rmm::mr::device_memory_resource* mr)
+{
+  using namespace numeric;
+  using Type                   = device_storage_type_t<T>;
+  using FixedPointRoundFunctor = RoundFunctor<Type>;
+
+  // if rounding to more precision that fixed_point is capable of, just need to rescale
+  // note: decimal_places has the opposite sign of numeric::scale_type (therefore have to negate)
+  if (is_fixed_point(input.type()) && input.type().scale() > -decimal_places) {
```

Comment:
Do we need to check `is_fixed_point()` here, since we're SFINAEd already?

Reply:
Great catch! This was leftover from a previous design - we definitely don't need it now! Fixed.